### ISSUE TITLE
[Support]: Stuck in restart loop

### ISSUE THREAD

**Issue by raintonr** [blakeblackshear/frigate#4414]
### Describe the problem you are having

Stuck in restart loop see log.

### Version

0.11.1-2eada21

### Frigate config file

```yaml
logger:
  default: info

mqtt:
  host: redacted

detect:
  width: 704
  height: 576
  fps: 3
  max_disappeared: 30

detectors:
  cpu1:
    type: cpu
    num_threads: 2

objects:
  track:
    - car
    - person

ffmpeg:
  hwaccel_args:
    -hwaccel vaapi
    -hwaccel_device /dev/dri/renderD128
    -hwaccel_output_format yuv420p

snapshots:
  enabled: True

record:
  enabled: True

live:
  height: 576

birdseye:
  enabled: False
  width: 704
  height: 576
  mode: continuous

cameras:
  porch_intercom:
    rtmp:
      enabled: false
    ffmpeg:
      inputs:
        - path: rtsp://redacted@redacted:554/Streaming/Channels/2
          roles:
            - detect
        - path: rtsp://redacted@redacted:554/Streaming/Channels/1
          roles:
            - record
    detect:
      enabled: true
      width: 704
      height: 576
    mqtt:
      enabled: false
    objects:
      mask: 35,499,131,488,136,350,76,345,42,372
      filters:
        car:
          mask:
            - 220,351,337,451,92,510,64,354
    motion:
      mask:
        - 638,462,473,462,473,443,638,443
```

### Relevant log output

```shell
# docker-compose up
[+] Running 1/1
 ⠿ Container frigate  Created                                                                                                                                      1.4s
Attaching to frigate
frigate  | [s6-init] making user provided files available at /var/run/s6/etc...exited 0.
frigate  | [s6-init] ensuring user provided files have correct perms...exited 0.
frigate  | [fix-attrs.d] applying ownership & permissions fixes...
frigate  | [fix-attrs.d] done.
frigate  | [cont-init.d] executing container initialization scripts...
frigate  | [cont-init.d] done.
frigate  | [services.d] starting services
frigate  | [services.d] done.
frigate  | [2022-11-16 19:29:40] frigate.app                    INFO    : Starting Frigate (0.11.1-2eada21)
frigate  | Starting migrations
frigate  | [2022-11-16 19:29:40] peewee_migrate                 INFO    : Starting migrations
frigate  | There is nothing to migrate
frigate  | [2022-11-16 19:29:40] peewee_migrate                 INFO    : There is nothing to migrate
frigate  | [2022-11-16 19:29:40] detector.cpu1                  INFO    : Starting detection process: 215
frigate  | [2022-11-16 19:29:40] frigate.app                    INFO    : Output process started: 217
frigate  | [2022-11-16 19:29:40] ws4py                          INFO    : Using epoll
frigate  | [2022-11-16 19:29:40] frigate.edgetpu                WARNING : CPU detectors are not recommended and should only be used for testing or for trial purposes.
frigate  | [2022-11-16 19:29:40] frigate.app                    INFO    : Camera processor started for porch_intercom: 221
frigate  | [2022-11-16 19:29:40] frigate.app                    INFO    : Capture process started for porch_intercom: 222
frigate  | [cont-finish.d] executing container finish scripts...
frigate  | [cont-finish.d] done.
frigate  | [s6-finish] waiting for services.
frigate  | [2022-11-16 19:29:41] ws4py                          INFO    : Using epoll
frigate  | [s6-finish] sending all processes the TERM signal.
frigate  | [s6-finish] sending all processes the KILL signal and exiting.
frigate exited with code 0
frigate  | [s6-init] making user provided files available at /var/run/s6/etc...exited 0.
frigate  | [s6-init] ensuring user provided files have correct perms...exited 0.
frigate  | [fix-attrs.d] applying ownership & permissions fixes...
frigate  | [fix-attrs.d] done.
frigate  | [cont-init.d] executing container initialization scripts...
frigate  | [cont-init.d] done.
frigate  | [services.d] starting services
frigate  | [services.d] done.
frigate  | [2022-11-16 19:29:49] frigate.app                    INFO    : Starting Frigate (0.11.1-2eada21)
frigate  | Starting migrations
frigate  | [2022-11-16 19:29:49] peewee_migrate                 INFO    : Starting migrations
frigate  | There is nothing to migrate
frigate  | [2022-11-16 19:29:49] peewee_migrate                 INFO    : There is nothing to migrate
frigate  | [2022-11-16 19:29:49] detector.cpu1                  INFO    : Starting detection process: 217
frigate  | [2022-11-16 19:29:49] frigate.app                    INFO    : Output process started: 219
frigate  | [2022-11-16 19:29:49] frigate.edgetpu                WARNING : CPU detectors are not recommended and should only be used for testing or for trial purposes.
frigate  | [2022-11-16 19:29:49] frigate.app                    INFO    : Camera processor started for porch_intercom: 223
frigate  | [2022-11-16 19:29:49] ws4py                          INFO    : Using epoll
frigate  | [2022-11-16 19:29:49] frigate.app                    INFO    : Capture process started for porch_intercom: 225
frigate  | [2022-11-16 19:29:49] ws4py                          INFO    : Using epoll
frigate  | [cont-finish.d] executing container finish scripts...
frigate  | [cont-finish.d] done.
frigate  | [s6-finish] waiting for services.
frigate  | [s6-finish] sending all processes the TERM signal.
frigate  | [s6-finish] sending all processes the KILL signal and exiting.
frigate exited with code 0
```

### FFprobe output from your camera

```shell
ffprobe version 5.0.1 Copyright (c) 2007-2022 the FFmpeg developers
  built with gcc 12 (GCC)
  configuration: --prefix=/usr --bindir=/usr/bin --datadir=/usr/share/ffmpeg --docdir=/usr/share/doc/ffmpeg --incdir=/usr/include/ffmpeg --libdir=/usr/lib64 --mandir=/usr/share/man --arch=x86_64 --optflags='-O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection' --extra-ldflags='-Wl,-z,relro -Wl,--as-needed -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -Wl,--build-id=sha1 ' --extra-cflags=' -I/usr/include/rav1e' --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libvo-amrwbenc --enable-version3 --enable-bzlib --enable-chromaprint --disable-crystalhd --enable-fontconfig --enable-frei0r --enable-gcrypt --enable-gnutls --enable-ladspa --enable-libaom --enable-libdav1d --enable-libass --enable-libbluray --enable-libbs2b --enable-libcdio --enable-libdrm --enable-libjack --enable-libfreetype --enable-libfribidi --enable-libgsm --enable-libilbc --enable-libmp3lame --enable-libmysofa --enable-nvenc --enable-openal --enable-opencl --enable-opengl --enable-libopenjpeg --enable-libopenmpt --enable-libopus --enable-libpulse --enable-librsvg --enable-librav1e --enable-librtmp --enable-librubberband --enable-libsmbclient --enable-version3 --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libsrt --enable-libssh --enable-libsvtav1 --enable-libtesseract --enable-libtheora --enable-libtwolame --enable-libvorbis --enable-libv4l2 --enable-libvidstab --enable-libvmaf --enable-version3 --enable-vapoursynth --enable-libvpx --enable-vulkan --enable-libglslang --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxvid --enable-libxml2 --enable-libzimg --enable-libzmq --enable-libzvbi --enable-lv2 --enable-avfilter --enable-libmodplug --enable-postproc --enable-pthreads --disable-static --enable-shared --enable-gpl --disable-debug --disable-stripping --shlibdir=/usr/lib64 --enable-lto --enable-libmfx --enable-runtime-cpudetect
  libavutil      57. 17.100 / 57. 17.100
  libavcodec     59. 18.100 / 59. 18.100
  libavformat    59. 16.100 / 59. 16.100
  libavdevice    59.  4.100 / 59.  4.100
  libavfilter     8. 24.100 /  8. 24.100
  libswscale      6.  4.100 /  6.  4.100
  libswresample   4.  3.100 /  4.  3.100
  libpostproc    56.  3.100 / 56.  3.100
[rtsp @ 0x55bc1ac34e40] Missing PPS in sprop-parameter-sets, ignoring
Input #0, rtsp, from 'rtsp://redacted@redacted:554/Streaming/Channels/2':
  Metadata:
    title           : Media Presentation
  Duration: N/A, start: 0.040000, bitrate: N/A
  Stream #0:0: Video: h264 (Baseline), yuvj420p(pc, bt709, progressive), 704x576, 25 fps, 25 tbr, 90k tbn
  Stream #0:1: Audio: pcm_mulaw, 8000 Hz, 1 channels, s16, 64 kb/s
```

### Operating system

Other Linux

### Install method

Docker Compose

### Coral version

CPU (no coral)

### Network connection

Wired

### Camera make and model

Hikvision DS-KD8003-IME1

### Any other information that may be helpful

Was working fine for ages but noticed MQTT not being updated and upon restart noticed this loop occuring.

**Comment by NickM-27** [comment]
This usually happens when frigate sees MQTT but is unable to write to it. Recommend restarting both frigate and MQTT at the same time

**Comment by raintonr** [comment]
The MQTT server is definitely available. In fact, in the MQTT server log I can actually see Frigate connecting and then disconnecting when it shuts down.

**Comment by NickM-27** [comment]
Not seeing any errors, not sure why this would be occurring, it is getting the stop command from somewhere

**Comment by raintonr** [comment]
Any way to turn on logging that would show that?

**Comment by NickM-27** [comment]
Not that I know of, here's another example of where MQTT could be connected to but frigate stopped immediately https://github.com/blakeblackshear/frigate/issues/3630

**Comment by NickM-27** [comment]
also one more instance where we saw this: https://github.com/blakeblackshear/frigate/issues/3399#issuecomment-1170021777

**Comment by raintonr** [comment]
I note one report of `--net=host` being of help so added `network_mode: "host"` to my `docker-compost.yml`

No change.

TBH, I don't see how this could be it as mentioned before, everything has been working fine for ages. Neither the MQTT or Docker host where Frigate runs have been restarted.

**Comment by NickM-27** [comment]
The thing is, frigate only has a few reasons to stop:
1. Detector is setup as coral and the coral is suddenly not found anymore.
2. MQTT issues.
3. The host stops it.

Of these, 1 always has a log along with a python crash log. 2 usually has logs but there are documented examples of other users which had MQTT issues and there was no error in the log, 3 also but assuming you didn't have this issue previously and nothing changed I wouldn't suspect it.

You also mentioned the MQTT broker having some issue and then this started, leading me toward that conclusion. In 0.12 we are looking to make MQTT optional and not stop frigate when there is an issue, due to these handful of occurrences.

**Comment by raintonr** [comment]
One thing maybe worth mentioning? Maybe a week or so ago I updated all my hosts to be IPv6 enabled (received a new ISP router that supports IPv6 so enabled in the LAN). However, this should be irrelevant as the MQTT host is configured using it's IPv4 address and as I said already, connectivity to that can be seen to be working.

**Comment by NickM-27** [comment]
> You also mentioned the MQTT broker having some issue and then this started...
> 
> No. I noticed that the MQTT topic hadn't been updated when it should have been and wondered why. Which led me to look at Frigate.
> 
> There was an 'in progress' even listed for the camera which should have stopped an hour or two earlier, so I deleted that event. Nothing happened. So I restarted Frigate and... noticed this loop.

I see, I misunderstood what that comment meant.  In any case, still lends itself to the idea that something might have happened in the MQTT broker that caused an issue since it was working until it wasn't.

In any case I don't currently have other ideas so will think on it or perhaps someone else will have an idea

**Comment by raintonr** [comment]
Ah-ha! What's this:

```
[2022-11-16 22:20:24] frigate.mqtt                   DEBUG   : Received mqtt message on frigate/restart.
```

This message doesn't appear to come out all the time. I found it in `journalctl -f -u docker`

I'm using [ioBroker](https://www.iobroker.net/) for the MQTT server. I have no idea where this `frigate/restart` topic came from or how it could have been updated but would that be causing the restart?

Either way, I stopped Frigate, stopped the MQTT server, deleted all old states (topics) on the MQTT server, restarted MQTT server and restarted Frigate and it's back working OK.

After this process, I do not see the `frigate/restart` topic on the MQTT server so where that came from is a mystery!
